### PR TITLE
Added tests and docs to the edge cases in the 'nth' function.

### LINKED ISSUE #137: Improve 'nth()' function tests and documentation
As mentioned #136 and #130 there're some edge cases which are not tested and documented. Especially differences between taking iterators and generators as the second argument

### PR DESCRIPTION
**Pull request by ma2ciek**
### Suggested merge commit message ([convention](https://github.com/ckeditor/ckeditor5-design/wiki/Git-commit-message-convention))

Internal: Added tests and docs to the edge cases in the 'nth' function. Closes #137.

### CODE REVIEW

File: tests/nth.js [hunk 1]
```diff
@@ -8,26 +8,43 @@ import nth from '../src/nth';
...
+		it( 'should stop inside the given iterator', () => {
+			const collection = [ 11, 22, 33 ];
+			const iterator = collection[ Symbol.iterator ]();
+
+			nth( 0, iterator );
+
+			expect( iterator.next().value ).to.equal( 22 );
+		} );
+
+		function *getGenerator() {
```
Comment (Reinmar) [inline tests/nth.js]: It is a generator which returns an iterator. So it should stay being `getIterator()`.

File: src/nth.js [hunk 2]
```diff
@@ -8,14 +8,16 @@
  */
 
 /**
- * Returns `nth` (starts from `0` of course) item of an `iterable`.
+ * Returns `nth` (starts from `0` of course) item of the given `iterable`.
+ * Consumes all items of the generator.
+ * Consumes all items up to the index of the iterator, including the returned item.
```
Comment (Reinmar) [inline src/nth.js]: This looks like a duplicated sentence. So how does it work finally?

Comment (ma2ciek) [inline src/nth.js]: it consumes the generator and it doesn't consume the iterator if the iterator doesn't implement the `return() method`, like e.g. Array Iterator. http://www.ecma-international.org/ecma-262/7.0/#sec-iteratorclose. I can add it to the comments to clarify what's going on.

File: tests/nth.js (hunk 1 above)
Comment (ma2ciek) [inline tests/nth.js]: Generator implements the iterator protocol, but the iterator and generator objects aren't the same and behave differently in some situations.

File: src/nth.js (hunk 2 above)
Comment (ma2ciek) [inline src/nth.js]: These tests are showing the difference: https://github.com/ckeditor/ckeditor5-utils/pull/138/files#diff-7b7a935b05c5a95c53b4e26cadf83040R30

File: tests/nth.js (hunk 1 above)
Comment (Reinmar) [inline tests/nth.js]: OK, I didn't know about that. So if you want to differentiate that, then the a better name would be `generateValues()` or `createGenerator()` **AND** it such things must be documented somewhere. Note that none of us would understand the real meaning of this change.

Comment (Reinmar) [inline tests/nth.js]: Hm... Or maybe I'm overexaggerating or even wrong. I checked that generator is an object, the function is called "generator function" (see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Generator). So you're right – it can be `getGenerator()` because it actually returns a generator. My bad.

Comment (Reinmar) [inline tests/nth.js]: Previously I thought that that function is a generator and it returns an iterator.

Comment (ma2ciek) [inline tests/nth.js]: For me both `getGenerator` and `createGenerator` seem good. `Generator` is a little bit unintuitive term.

Comment (ma2ciek) [inline tests/nth.js]: > Previously I thought that that function is a generator and it returns an iterator.

Comment (Reinmar) [inline tests/nth.js]: Let's keep `getGenerator()`. After I understood the proper naming of these objects it all makes sense to me.

**Comment by ma2ciek** [comment]
Docs for the `nth` are done.
